tests/e2e/contexts.ts

```diff
@@ -0,0 +1,91 @@
+import { MongoClient } from 'mongodb';
+import { MongoMemoryReplSet } from 'mongodb-memory-server';
+import { MaggregorProcess } from './setup-maggregor';
+import { startMongoServer, loadTestData, healthCheck } from './utils';
+
+const testAtlasUsername = process.env.TEST_ATLAS_USERNAME;
+const testAtlasPassword = process.env.TEST_ATLAS_PASSWORD;
+
+if (!testAtlasUsername || !testAtlasPassword) {
+  throw new Error(
+    'E2E Tests cannot run on Mongo Atlas without TEST_ATLAS_USERNAME and TEST_ATLAS_PASSWORD variables.',
+  );
+}
+
+export type CommonContext = {
+  type: string;
+  name: string;
+  ssl?: boolean;
+  maggregor?: MaggregorProcess;
+  maggreUri?: string;
+  maggreClient?: MongoClient;
+  mongoUri?: string;
+  mongoClient?: MongoClient;
+};
+
+export type RemoteContext = {
+  type: 'remote';
+  externalMongoUri: string;
+} & CommonContext;
+
+export type LocalContext = {
+  type: 'local';
+  startMongo: boolean;
+  // If startMongo is false, then mongoUri must be required
+  mongoUri?: string;
+} & CommonContext;
+
+export type Context = RemoteContext | LocalContext;
+
+export const contexts: Context[] = [
+  {
+    type: 'remote',
+    name: 'Remote on shared MongoDB Atlas v6.0.0',
+    ssl: true,
+    externalMongoUri: `mongodb://${testAtlasUsername}:${testAtlasPassword}@ac-edqjyiy-shard-00-02.puyjt1x.mongodb.net:27017/?ssl=true&replicaSet=atlas-9duftq-shard-0&retryWrites=true&w=majority`,
```

Comment:
the hostname should also be an env variable in my opinion

Reply:
There is also the replicatSet. So yes maybe just use a variable for the entire url